sql/2020/05_Media/meta_open_image_types.sql

```diff
@@ -0,0 +1,48 @@
+#standardSQL
+# meta open graph image types
+
+# returns all the data we need from _almanac
+CREATE TEMPORARY FUNCTION get_almanac_info(almanac_string STRING)
+RETURNS STRUCT<
+  image_types ARRAY<STRING>
+> LANGUAGE js AS '''
+var result = {};
+try {
+    var almanac = JSON.parse(almanac_string);
+
+    if (Array.isArray(almanac) || typeof almanac != 'object') return result;
+
+    for (var node of almanac['meta-nodes']['nodes']) {
+      if (node['property'] === 'og:image') {
+        if (result.image_types == null)
+        result.image_types = [];
+        let url = node['content'];
+        var rex = new RegExp('\\.([^/#?]+)[.]([^/#?]+)([#?][^/]*)?$');
```

Comment:
I'm having trouble understanding this regex. While I'm sure it's correct, could we have a comment above it explaining it?